modules/mauth-common/src/main/java/com/mdsol/mauth/MAuthRequest.java

```diff
@@ -160,20 +188,36 @@ public Builder withQueryParameters(String queryParameters) {
       return this;
     }
 
-    public Builder withRequestHeaders(Map<String, String> headers) {
-      this.headers = headers;
+    /**
+     * Set Mauth headers (it may include the both sets of Mauth V1 and V2)
+     * @param mauthHeaders the request headers for Mauth, such as
+     *        Mauth V1, X_MWS_AUTHENTICATION_HEADER_NAME, X_MWS_TIME_HEADER_NAME
+     *        Mauth V2, MCC_AUTHENTICATION_HEADER_NAME and MCC_TIME_HEADER_NAME
+     *
+     * @return
+     */
+    public Builder withMauthHeaders(Map<String, String> mauthHeaders) {
+      this.mauthHeaders = mauthHeaders;
       return this;
     }
 
+    /**
+     * Construct a MAuthRequest object
+     *
+     * @Note
+     * If mauthHeaders are provided, get the value of the highest protocol version to construct object
+     *
+     * @return a object of MAuthRequest
+     */
     public MAuthRequest build() {
       // get the newest mauth version from the request headers...
-      if (headers != null && !headers.isEmpty()) {
-        if (headers.get(MCC_AUTHENTICATION_HEADER_NAME) != null) {
-          authenticationHeaderValue = headers.get(MCC_AUTHENTICATION_HEADER_NAME);
-          timeHeaderValue = headers.get(MCC_TIME_HEADER_NAME);
+      if (mauthHeaders != null && !mauthHeaders.isEmpty()) {
+        if (mauthHeaders.get(MCC_AUTHENTICATION_HEADER_NAME) != null) {
+          authenticationHeaderValue = mauthHeaders.get(MCC_AUTHENTICATION_HEADER_NAME);
+          timeHeaderValue = mauthHeaders.get(MCC_TIME_HEADER_NAME);
         } else {
-          authenticationHeaderValue = headers.get(X_MWS_AUTHENTICATION_HEADER_NAME);
-          timeHeaderValue = headers.get(X_MWS_TIME_HEADER_NAME);
+          authenticationHeaderValue = mauthHeaders.get(X_MWS_AUTHENTICATION_HEADER_NAME);
+          timeHeaderValue = mauthHeaders.get(X_MWS_TIME_HEADER_NAME);
```

Comment:
Header extraction logic needs to be case insensitive.

Reply:
mauthHeaders is Map<String, String> to store mauth headers provided by consumer and have mentioned the supported mauth headers in readme for Java user. I can change it to new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER)

Reply:
yeap sounds good. Make `withMauthHeaders` take a Map<String, String> and insert it into `this.mauthHeaders` which is a TreeMap

Reply:
Updated and added a couple specs to verify https://github.com/mdsol/mauth-jvm-clients/pull/92/commits/735bd0daeb6afd8ec05d1562413a7721f033797f

Reply:
Nicely done